src/main/java/models/Song.java

```
@@ -0,0 +1,88 @@
+package models;
+
+import exception.IncorrectParametersException;
+
+/**
+ * A class that represent musical composition intended to be performed by the performer.
+ *
+ * The Song has methods for validation about title and length,
+ * also methods for getting the information about the song.
+ */
+public class Song {
+    private final String title;
+    private final Performer performer;
+    private final Genre genre;
+    private final int lengthInSeconds;
+
+    public Song(String title, Performer performer, Genre genre, int lengthInSeconds) throws IncorrectParametersException {
+        title = validateParameters(title, lengthInSeconds);
+
+        this.title = title;
+        this.performer = performer;
+        this.genre = genre;
+        this.lengthInSeconds = lengthInSeconds;
+    }
+
+    public String getTitle() {
+        return title;
+    }
+
+    public String getPerformerName() {
+        return performer.getName();
+    }
+
+    public int getLengthInSeconds() {
+        return lengthInSeconds;
+    }
+
+    /**
+     * @param title -> represent song's title
+     * @param lengthInSeconds -> represent song's length
+     * @return song's title after validation
+     * @throws IncorrectParametersException when length is invalid
+     */
+    private String validateParameters(String title, int lengthInSeconds) throws IncorrectParametersException {
+        if (validateTitle(title)) {
+            title = "No title";
+        }
+        if (validateSongLength(lengthInSeconds)) {
+            throw new IncorrectParametersException("Incorrect song's length");
```

you can throw this exception in validateSongLength()